auth-server/src/services/auth.ts

```diff
@@ -0,0 +1,126 @@
+import * as jwt from 'jsonwebtoken'
+import { NotFoundError } from 'objection'
+import { DataError } from 'objection-db-errors'
+import { uuid } from 'uuidv4'
+
+import User from '../model/User'
+import RefreshToken from '../model/RefreshToken'
+
+export default class AuthService {
+	constructor(){}
+
+	public async Login(username, password): Promise<any> {
+		const user = await User
+			.query()
+			.where('username', username)
+			.first()
+      
+		if (!user) { 
+			throw new NotFoundError('User not found')
+		}
+
+		const correctPassword = await user.verifyPassword(password)
+		if (!correctPassword) {
+			throw new DataError('Incorrect password')
+		}
+
+		return {
+			user: {
+				email: user.email,
+				username: user.username,
+				name: user.name
+			},
+			token: this.getSignedToken(user),
+			refreshToken: await this.getRefreshToken(user)
+		}
+	}
+
+	public async SignUp(email, password, username, name): Promise<any> {
+		const user = await User.query()
+			.allowInsert('[email, password, username, name]')
+			.insert({
+				email,
+				password,
+				username,
+				name
+			})
+		
+		return {
+			user: {
+				email: user.email,
+				username: user.username,
+				name: user.name
+			},
+			token: this.getSignedToken(user),
+			refreshToken: await this.getRefreshToken(user)
+		}
+	}
+
+	public async RefreshToken(token: string) {
+		const refreshToken = await RefreshToken
+			.query()
+			.where('token', token)
+			.first()
+		
+		if (!refreshToken) { 
+			throw new NotFoundError('Refresh token not found')
+		}
+
+		if (!refreshToken.valid) {
+			throw new Error('Refresh token not valid')
+		}
+
+		if (refreshToken.expires < Math.floor(Date.now() / 1000)) {
+			throw new Error('Refresh token expired')
+		}
+
+		const user = await User
+			.query()
+			.where('id', refreshToken.user_id)
+			.first()
+
+		return this.getSignedToken(user)
+	}
+
+	private getSignedToken({ id, username, email }) {
+		const allowedRoles = ['user']
+		let currentRole = 'user'
+	
+		// if our user is the proposal bot, give additional role.
+		if (id == process.env.BOT_PROPOSAL_USER_ID) { // eslint-disable-line
+			allowedRoles.push('proposal_bot')
+			currentRole = 'proposal_bot'
+		}
+	
+		const tokenContent = {
+			sub: `${id}`,
+			name: username,
+			iat: Math.floor(Date.now() / 1000),
+			'https://hasura.io/jwt/claims': {
+				'x-hasura-allowed-roles': allowedRoles,
+				'x-hasura-default-role': currentRole,
+				'x-hasura-user-id': `${id}`,
+				'x-hasura-user-email': email
+			}
+		}
+
+		return jwt.sign(tokenContent, process.env.ENCRYPTION_KEY, { expiresIn: '10d' })
```

Comment:
I'd go for `1h` here to follow what [I found on the internet](https://blog.cloud-elements.com/oauth-2-0-access-refresh-token-guide)

Reply:
done